lua/neoterm.lua

```diff
@@ -30,21 +32,63 @@ local neoterm = {}
 
 -- Sets global configuration
 -- Options:
---	clear_on_run - send the clear comand before running a command with run or rerun
---	mode - set how the terminal window will be displayed
---		* vertical
---		* horizonal
---		* fullscreen
```

Comment:
To save a breaking change, could we set the `position`, `width`, and `height` based on this parameter? i.e. `horizontal` translates to `position = 'right', `vertical` translates to `position = 'bottom'`, and `fullscreen` translates to `position = 'fullscreen'`

If the `position` is set, it should take priority and `mode` should be ignored.

Could add a deprecated message to the mode param.

Reply:
Added a deprecated warning interrupting the user if the mode option is set.

Also 33326e418501c7e0c933981083663bc012634114 adds a simple translation from the old mode option to the newer position option